staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3068,6 +3068,50 @@ type ContainerStatus struct {
 	// +featureGate=SupplementalGroupsPolicy
 	// +optional
 	User *ContainerUser `json:"user,omitempty" protobuf:"bytes,13,opt,name=user,casttype=ContainerUser"`
+	// AllocatedResourcesStatus represents the status of various resources
+	// allocated for this Pod.
+	// +featureGate=ResourceHealthStatus
+	// +optional
+	AllocatedResourcesStatus ResourcesStatus `json:"allocatedResourcesStatus,omitempty" protobuf:"bytes,14,rep,name=allocatedResourcesStatus,casttype=ResourcesStatus,castkey=ResourceName"`
```

Comment:
Should this be a pointer?

Reply:
I am not sure we instroduce some fields without the pointer. Example: https://github.com/kubernetes/kubernetes/blob/d21b17264e5a554724aa3ad032536630bcfd5b3f/pkg/apis/core/types.go#L674

Reply:
I belive this applies? 

"Optional fields have the following properties... They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) or have a built-in nil value (e.g. maps and slices)." https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
oh, wait, I should have read further.  It's an alias to a Map so it's already nilable.

Reply:
Resolving as the type is nullable. Thank you!